os/arch/arm/src/s5j/Make.defs

```diff
@@ -109,6 +109,14 @@ ifeq ($(CONFIG_BUILD_KERNEL),y)
 CMN_CSRCS += up_task_start.c up_pthread_start.c arm_signal_dispatch.c
 endif
 
+ifeq ($(CONFIG_BUILD_PROTECTED),y)
+CMN_CSRCS += up_task_start.c up_pthread_start.c
+ifneq ($(CONFIG_DISABLE_SIGNALS),y)
+CMN_CSRCS += arm_signal_dispatch.c
+CMN_UASRCS += arm_signal_handler.S
+endif
+endif
+
```

Comment:
If possible handle files common to CONFIG_BUILD_KERNEL  and  CONFIG_BUILD_PROTECTED  under single  build path